docs/03-language-guide/14-equality.md

```diff
@@ -1,6 +1,133 @@
 ---
 title: Equality
 id: equality
+description: A reference guide to equality in Wing.
+keywords: [equality, equals, identity]
 ---
 
-TODO (for now, see the [language spec RFC](../999-contributors/999-rfcs/2023-06-12-language-spec.md))
+Checking for equality is performed with the `==` operator. It returns `true` if the two values are equal, and `false` otherwise.
+
+The main difference between equality in JavaScript and Wing is that `==` in Wing is not allowed to compare values of different types. For example, `1 == "1"` is not allowed in Wing, and will result in a compile-time error.
+
+The execution phase ([preflight or inflight](../02-core-concepts/01-preflight-and-inflight.md)) that a value was created in does not affect its equality. For example, a value created in preflight can be equal to a value created in inflight.
+
+Equality in Wing is a symmetric and transitive relationship - that is, (1) if `a == b`, then `b == a`, and (2) if `a == b` and `b == c`, then `a == c`.
+
+The following is a set of rules for checking equality:
+
+## Basic types
+
+1. Two `str` values are equal if they have the same characters in the same order.
+2. Two `num` values are equal if they have the same floating-point value. The [IEEE 754] standard is used for storing numbers, which means that for example `-0 == +0`. `NaN` is not equal to any value, including itself.
+3. Two `bool` values are equal if they are both `true` or both `false`.
+4. Two `duration` values are equal if they have the same number of milliseconds.
+5. Two `T?` types (optional `T` values) are equal if they are both empty (`nil`) or both non-empty, and if they are both non-empty, their inner values are equal. A value of type `T?` is never equal to a value of type `T`.
```

Comment:
Something that came up [here](https://github.com/winglang/wing/pull/2918#discussion_r1230673067) is that maybe we should allow comparing `T?` and `T`:

```js
let var x: str? = "hello";

if x == "hello" {
  log("x is hello");
}

x = nil;
if x != "hello" {
  log("x is not hello");
}
```

Reply:
I'm for staying explicit about this for now. But I also see the value (pun intended) of implicitly checking for nil.
If we take that course there's the question of being recursive about it. Can we compare a `T???` with a `T`? Can we compare a `T??` with a `T?`? I think we should allow all of these if we allow any of them.

Reply:
> I'm for staying explicit about this for now.

What's the motivation?

Reply:
> What's the motivation?

I think the idea is to require unwrapping optional values, no?